src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -1511,70 +1604,77 @@ private CloudApplication findApplicationByName(String name, boolean required) {
     }
 
-    private Flux<? extends Resource<ApplicationEntity>> getApplicationResources() {
-        IntFunction<ListSpaceApplicationsRequest> pageRequestSupplier = page -> ListSpaceApplicationsRequest.builder()
-                                                                                                            .spaceId(getTargetSpaceGuid().toString())
-                                                                                                            .page(page)
-                                                                                                            .build();
-        return PaginationUtils.requestClientV2Resources(page -> delegate.spaces()
-                                                                        .listApplications(pageRequestSupplier.apply(page)));
+    private Flux<? extends Application> getApplicationResources() {
+        IntFunction<ListApplicationsRequest> pageRequestSupplier = page -> ListApplicationsRequest.builder()
+                                                                                                  .spaceId(getTargetSpaceGuid().toString())
+                                                                                                  .page(page)
+                                                                                                  .build();
+        return PaginationUtils.requestClientV3Resources(page -> delegate.applicationsV3()
+                                                                        .list(pageRequestSupplier.apply(page)));
     }
 
-    private Mono<? extends Resource<ApplicationEntity>> getApplicationResource(UUID guid) {
-        GetApplicationRequest request = GetApplicationRequest.builder()
-                                                             .applicationId(guid.toString())
-                                                             .build();
-        return delegate.applicationsV2()
-                       .get(request);
+    private Mono<? extends Application> getApplicationResource(UUID guid) {
+        IntFunction<ListApplicationsRequest> pageRequestSupplier = page -> ListApplicationsRequest.builder()
+                                                                                                  .spaceId(getTargetSpaceGuid().toString())
+                                                                                                  .applicationId(guid.toString())
+                                                                                                  .page(page)
+                                                                                                  .build();
+        return PaginationUtils.requestClientV3Resources(page -> delegate.applicationsV3()
+                                                                        .list(pageRequestSupplier.apply(page)))
+                              .singleOrEmpty();
```

Comment:
If we want the application by the given Guid, why can't we use the `get` request?

Reply:
Done